materialize-mysql/client.go

```diff
@@ -99,7 +103,89 @@ func (c *client) InfoSchema(ctx context.Context, resourcePaths [][]string) (is *
 	return sql.StdFetchInfoSchema(ctx, c.db, c.ep.Dialect, "def", resourcePaths)
 }
 
+func formatOffset(loc *time.Location) string {
+	now := time.Now().In(loc)
+	_, offset := now.Zone()
+
+	sign := "+"
+	if offset < 0 {
+		sign = "-"
+		offset = -offset
+	}
+	hours := offset / 3600
+	minutes := (offset % 3600) / 60
+	return fmt.Sprintf("%s%02d:%02d", sign, hours, minutes)
+}
+
+func castRfc3339ToDateTime(loc *time.Location, identifier string) string {
+	return fmt.Sprintf("CONVERT_TZ(STR_TO_DATE(SUBSTRING(%s, 1, 19), '%%Y-%%m-%%dT%%H:%%i:%%s'), '+00:00', '%s')", identifier, formatOffset(loc))
+}
+
+func (c *client) columnMigrationSteps(ctx context.Context) []sql.ColumnMigrationStep {
+	return []sql.ColumnMigrationStep{
+		func(dialect sql.Dialect, table sql.Table, migration sql.ColumnTypeMigration, firstTempColumnIdentifier, _ string) (string, error) {
+			return fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s;",
+				table.Identifier,
+				firstTempColumnIdentifier,
+				migration.DDL,
+			), nil
+		},
+		func(dialect sql.Dialect, table sql.Table, migration sql.ColumnTypeMigration, firstTempColumnIdentifier, _ string) (string, error) {
+			// Migrating from string to datetime
```

Comment:
@williamhbaker let's discuss this in Slack with the rest of the team. I mainly wanted to show that supporting such cases would also be supported with the current factoring, I'd also be happy to remove this bit of code since it can be problematic